pkg/virt-handler/container-disk/mount.go

```diff
@@ -236,7 +236,7 @@ func (m *mounter) Mount(vmi *v1.VirtualMachineInstance, verify bool) error {
 				if err != nil {
 					return fmt.Errorf("failed to detect root mount point of containerDisk %v on the node: %v", volume.Name, err)
 				}
-				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.Root, nodeMountInfo.MountPoint), volume.ContainerDisk.Path)
+				sourceFile, err := containerdisk.GetImage(filepath.Join(nodeRes.MountRoot(), nodeMountInfo.Root, nodeMountInfo.MountPoint, mountInfo.Root), volume.ContainerDisk.Path)
```

Comment:
can we get an explanation of how this fixes the issue?

Reply:
Ah, yeah, sorry, I mentioned it only in the linked issue: https://github.com/kubevirt/kubevirt/issues/4613#issuecomment-744489258 . In the scenario when `devicemapper` backend is used, the container filesystem gets mounted at `/var/lib/docker/devicemapper/mnt/<ID>/rootfs`. Previously the last part of the path (i.e. `/rootfs`) was missing and that was causing errors durinng VM startup:

```
open /proc/1/root/var/lib/docker/devicemapper/mnt/f98b595a54f0d1deec6b86c21996eacd3ce7cf3963d123d7dd0af9d59869f40f/disk: no such file or directory
```

According to my findings `mountInfo.Root` contains the needed rootfs mount info. Appending it to the base path fixes the issue.

Reply:
@vasiliy-ul that sounds like  the perfect PR description. Would you mind updating it with this content?

Reply:
And do you have a mount-table which you could transform into a unit-test?

Reply:
@rmohr , I think I can get the mount table and use it in the test. I assume it is the `virt-handler` pod I should look into, right?

Reply:
virt-handler looks it up from the host perspective (/proc/1/mountinfo) and from the virt-launcher perspective `/proc/<launcherpid>/mountinfo`. Both are needed.

This part is not yet really unit-test friendly. If it ends up too much work, it would be great to at least get dumps of the `mountinfo` files so that we can add the tests independently.

Reply:
Hm... yeah, maybe you are right about unit-test friendliness :) From what I see the functionality is spread across several modules: 1) `isolation.go` does the parsing of the mountinfo; 2) `mount.go` contains the logic of composing the base path; 3) `container-disk.go` builds the final disk path. In my understanding it is needed to cover with tests all the modules independently plus finally to provide a 'bigger' test for the complete flow. I would say that it will require some ammount of refactoring. At least for me it does not seem really straightforward now (but maybe because I am not that familiar with the complete codebase yet).

I have the dumps. Just not sure where to put those. In fact it is pretty easy to get those locally. Just a few steps are needed:

- `cluster-up` and `ssh` into the node
- `yum install lvm2 device-mapper-persistent-data` && systemctl stop docker
- add `"storage-driver": "devicemapper"` to `/etc/docker/daemon.json`
- `systemctl start docker`
- `cluster-sync` && `kubectl apply -f VMI`

Reply:
> I have the dumps. Just not sure where to put those. In fact it is pretty easy to get those locally. Just a few steps are needed:

That is really easy. :+1: 
Then never mind.